src/main/java/ladder/domain/Line.java

```diff
@@ -1,47 +1,56 @@
 package ladder.domain;
 
-import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.stream.Collectors;
 import java.util.stream.IntStream;
 
 public class Line {
-    private final List<Bar> bars;
-
-    public Line() {
-        bars = new ArrayList<>(List.of(new Bar(false)));
-    }
-
-    public void addBars(int peopleSize, BarGenerator barGenerator) {
-        IntStream.range(0, peopleSize - 1)
-                .forEach(count -> addCorrectBar(barGenerator));
-    }
-
-    private void addCorrectBar(BarGenerator barGenerator) {
-        if (lastBar().isExistBar()) {
-            addNotExistedBar();
-            return;
-        }
-
-        addBar(barGenerator);
-    }
-
-    private Bar lastBar() {
-        return bars.get(barsLastIndex());
-    }
-
-    private int barsLastIndex() {
-        return bars.size() - 1;
-    }
-
-    private void addBar(BarGenerator barGenerator) {
-        bars.add(new Bar(barGenerator));
-    }
-
-    private void addNotExistedBar() {
-        bars.add(new Bar(false));
-    }
-
-    public List<Bar> getBars() {
-        return bars;
+    private static final int BOTH_ENDS_COUNT = 2;
+    
+    private final List<Direction> line;
+
+    public Line(BarGenerator barGenerator, int peopleSize) {
+        line = createLine(barGenerator, peopleSize);
+    }
+    
+    private List<Direction> createLine(BarGenerator barGenerator, int peopleSize) {
+        LinkedList<Direction> line = new LinkedList<>();
+        addFirstDirection(barGenerator, line);
+        addMiddleDirections(barGenerator, peopleSize, line);
+        addLastDirection(line);
+        return line;
+    }
+    
+    private void addFirstDirection(BarGenerator barGenerator, LinkedList<Direction> line) {
+        line.add(Direction.createFirst(barGenerator));
+    }
+    
+    private void addMiddleDirections(BarGenerator barGenerator, int peopleSize, LinkedList<Direction> line) {
+        IntStream.range(0, peopleSize - BOTH_ENDS_COUNT)
+                .mapToObj(middleDirectionCount -> getLastDirection(line).createNext(barGenerator))
+                .forEach(line::add);
```

Comment:
```suggestion
        for (int i = 0; i < peopleSize - BOTH_END_COUNT; i++) {
          line.add(getLastDirection(line).createNext(barGenerator));
        }
```
2개 중에 코드를 처음 보는 사람에게는 어떤게 더 읽기 쉬운 코드인 것 같으신가요?ㅎㅎ

Reply:
제가 너무 for문보다 Stream 이 더 깔끔한 코드라는 편견을 가지고 있었던 것 같습니다. :smiling_face_with_tear:

Reply:
제가 너무 **'for문보다 Stream 이 더 깔끔한 코드라는 편견'** 을 가지고 있었던 것 같습니다. :sweat_smile:

이렇게 두개를 놓고 비교해보니 더 와닿네요. 리팩토링 하겠습니다.